sdks/java/core/src/main/java/org/apache/beam/sdk/util/RowJsonValueExtractors.java

```diff
@@ -177,6 +178,18 @@
         .build();
   }
 
+  /**
+   * Extracts DateTime from the JsonNode if it is valid.
+   *
+   * <p>Throws {@link UnsupportedRowJsonException} if value is out of bounds.
+   */
+  static ValueExtractor<DateTime> datetimeValueExtractor() {
+    return ValidatingValueExtractor.<DateTime>builder()
+        .setExtractor(jsonNode -> DateTime.parse(jsonNode.textValue()))
```

Comment:
What DateTime formats does this support? Is it just ISO-8601? Please document that in the docstring. Also if other formats are supported we should test them specifically.

Reply:
It currently only support ISO 8601 (the same format we write to PubSub). We can add support for more formats in the future. Clarified in the comment.